src/sqs/discovery.ts

```diff
@@ -1,12 +1,18 @@
-import {MessageHandler} from "./index"
+import { MessageHandler } from "./index";
 import app from "../worker/app";
-import {discovery} from "../sync/discovery";
+import { discovery, discoveryOctoKit } from "../sync/discovery";
+import { booleanFlag, BooleanFlags } from "../config/feature-flags";
 
 export type DiscoveryMessagePayload = {
 	installationId: number,
 	jiraHost: string
 }
 
 export const discoveryQueueMessageHandler : MessageHandler<DiscoveryMessagePayload> = async (context) => {
-	await discovery(app)({data: context.payload}, context.log);
-}
+	const useNewGHClient = await booleanFlag(BooleanFlags.USE_NEW_GITHUB_CLIENT_FOR_DISCOVERY, false, context?.payload?.jiraHost);
+	if(useNewGHClient) {
+		await discovery({ data: context.payload }, context.log);
```

Comment:
/nit we don't need to pass payload like "data" parameter. We can put it directly to the method parameter:
`await discovery(context.payload, context.log)`

`export const discovery = async (data: DiscoveryMessagePayload, logger: LoggerWithTarget) => {`
etc.

Reply:
ooo yeah, agreed, nice n tidy